plugins/modules/system/disk_facts.py

```diff
@@ -0,0 +1,181 @@
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: disk_facts
+version_added: "2.5.0"
+short_description: Get disk facts in the linux OS
+description:
+    - This module is to get disk facts like free space, used percentage, mount location etc in linux Ansible controller/controlled machines.
+requirements:
+  - psutil(python module)
+options:
+    name:
+        description:
+            - This is the disk name to query
+        required: false
+        default: all
+        type: str
+    filter:
+        description:
+            - This is the parameter used to filter the output
+        required: false
+        type: list
+        elements: str
+author:
+    - Saranya Sridharan(@saranyasridharan), Swetha M, Nagendra Kini, Anup Kumar, Mehmoodkha Pathan
+'''
+
+EXAMPLES = '''
+- name: Get disk facts
+  community.general.disk_facts:
+
+- name: Get Disk facts for Particular disk info
+  community.general.disk_facts:
+    name: diskname
+
+- name: Filter the output parameters for specific disk
+  community.general.disk_facts:
+    name: diskname
+    filter:
+      - mountpoint
+      - freespace
+      - totalsize
+
+- name: Filter the output parameters for all disk
+  community.general.disk_facts:
+    filter:
+      - fstype
+      - usedspace
+      - capacity_percentage
+'''
+
+
+RETURN = '''
+ansible_facts:
+    description: disk information only filtered parameter values if filter option is given.
+    returned: dictionary of disks informations
+    type: dict
+    sample: {
+        '/dev/disk1s1': {
+             'capacity_percentage': '2%'
+        }
+    }
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    import psutil
+    HAS_PSUTIL = True
+except ImportError:
+    HAS_PSUTIL = False
+
+
+# To calculate the disk size
+def calculate_size(bytes, suffix="B"):
+    """
+    Scale bytes to its proper format
+    e.g:
+        1253656 => '1.20MB'
+        1253656678 => '1.17GB'
+    """
+    factor = 1024
+    for unit in ["", "K", "M", "G", "T", "P"]:
+        if bytes < factor:
+            return "{0:.2f}{1}{2}".format(bytes, unit, suffix)
+        bytes /= factor
+
+
+# To get the disk partitions information
+def disk_facts(devicename):
+    output = {}
+    partitions = psutil.disk_partitions()
+    found = False
+    for partition in partitions:
+        if partition.device == devicename:
+            found = True
+        output[partition.device] = {}
+        output[partition.device]['mountpoint'] = partition.mountpoint
+        output[partition.device]['fstype'] = partition.fstype
+        try:
+            partition_usage = psutil.disk_usage(partition.mountpoint)
+        except PermissionError:
+            # this can be catched due to the disk that
+            # isn't ready
+            continue
+        output[partition.device]['totalsize'] = calculate_size(partition_usage.total)
```

Comment:
While these values are human readable, it is not really possible to process them in an efficient way. I would rather return them as integers, and allow users of the module to process them in whatever format they want.

Reply:
Now you have to document what unit they return. You also should document what is returned at all. Right now you only have a few examples and none of them processes the result, and users have to read the module source code (!) to find out what exactly is returned.